File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/AssignmentSubmissionResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AssignmentSubmissionResponse {

    private Integer id;

    private Integer assignmentId;

    private String assignmentTitle;

    private String idUser;

    private String userName;

    private String submissionText;

    private List<String> submissionFiles;

    private String submissionLink;

    private Date submittedAt;

    private Double score;

    private String feedback;

    private Date gradedAt;

    private String status;
}

File: react-type-vite/src/services/api/response/assignmentSubmissionResponse.ts
export interface AssignmentSubmissionResponse {
  id: number
  assignmentId: number
  assignmentTitle: string
  assignmentDescription?: string
  maxScore?: number
  deadline?: string
  idUser: string
  userName?: string
  studentName?: string
  email?: string
  submissionText?: string
  submissionFiles?: string[]
  submissionLink?: string
  submittedAt: string
  score?: number
  feedback?: string
  gradedAt?: string
  status: 'DRAFT' | 'SUBMITTED' | 'LATE' | 'GRADED'
  isLate?: boolean
  hoursLate?: number
}